tests/mock-git-annex

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+#
+# Helper for simulating of the git-annex external remote protocol.
+# Starts git-annex-remote-rclone and checks its behaviour against a script
+# read from stdin. Each line of the script is either:
+#
+# > COMMAND ARG ARG
+#     A literal command to send to git-annex-remote-rclone.
+#
+# < REGEX
+#     A grep -E compatible regex to match against a line read from
+#     git-annex-remote-rclone. DEBUG lines are ignored.
+#     If the match fails, the simulator exits with exit code 1.
+#
+# Any other kind of line makes the simulator exit with exit code 2.
+
+set -e
+
+me=$(basename -- "$0")
+cd "$(mktemp -d ${TMPDIR:-/tmp}/garr-XXXXXXX)"
```

Comment:
```suggestion
cd "$(mktemp -d \"${TMPDIR:-/tmp}/garr-XXXXXXX)\""
```
or I guess (didn't try)
```suggestion
cd "$(mktemp -d '${TMPDIR:-/tmp}/garr-XXXXXXX)'"
```
should work as well (since we already are within `""`), to please shellcheck?

```
In tests/mock-git-annex line 20:
7
cd "$(mktemp -d ${TMPDIR:-/tmp}/garr-XXXXXXX)"
8
                ^-- SC2086: Double quote to prevent globbing and word splitting.
```

Reply:
I took this from `tests/all-in-one.sh` but apparently that one is not shellchecked. Fixed it here.

Reply:
oh well -- we are working on "self-improvement" all the time -- much easier to identify shortcomings elsewhere ;)  Not yet sure why `all-in-one.sh` was not yet made shellcheck-ready. Thanks! I see that you nested `"` within -- and `shellcheck` is silent, so must be legit!  I trust nobody, so decided to check

```
❯ cat /tmp/spaces
#!/bin/bash

set -eu

cd "$(mktemp -d "${TMPDIR:-/tmp}/garr-XXXXXXX")"

pwd

❯ mkdir "/tmp/s p"
❯ TMPDIR="/tmp/s p" /tmp/spaces
/tmp/s p/garr-sENJsHg
```

so works as desired!  I will propose similar one to all-in-one.sh now